server/handlers.go

```diff
@@ -393,6 +440,8 @@ func ActionHandler(w http.ResponseWriter, r *http.Request) {
 
 // TFCHandler registers given record in local TFC
 func TFCHandler(w http.ResponseWriter, r *http.Request) {
+	// allow cross domain AJAX requests
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Why do you need cross domain AJAX requests? Could you please provide more info in comment, including examples. It is security flaw and we should remove this.

Reply:
Valentin,
I am allowing cross domain AJAX requests on few endpoints so that anyone can run our agent on any machine and can place a request or register new data to the main-agent.  

So, let's say main-agent is running on localhost:8989 and you are running your agent on localhost:8000. Now in order to register new transfer request, you can access HTML pages of your agent through this link - http://localhost:8000/html/client.html and can send HTTP post request to main-agent(localhost:8989) to register new request.